main.rb

```diff
@@ -0,0 +1,144 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    arr = to_a
+    while i < arr.length
+      yield(arr[i])
+      i += 1
+    end
+    self
+  end
+  [1, 2, 3].my_each { |n| puts "#{n}!" }
+
+  def my_each_each_with_index
```

Comment:
Please change this method's name to the correct name which is `my_each_with_index`.